nncf/quantization/algorithms/weight_compression/algorithm.py

```diff
@@ -360,16 +494,24 @@ def apply(
         statistic_points: Optional[StatisticPointsContainer] = None,
         dataset: Optional[Dataset] = None,
     ) -> TModel:
-        self._set_backend_entity(model)
-        nodes_to_compress = self._get_nodes_to_compress(graph)
-
+        self.set_backend_entity(model)
+        nodes_to_compress = self.get_nodes_to_compress(graph)
+        is_statistics_provided = statistic_points is not None
         statistics = None
-        data_aware_mixed_precision = (
-            self._sensitivity_metric != SensitivityMetric.WEIGHT_QUANTIZATION_ERROR and self._ratio != 1.0
-        )
-        data_aware_compression = self._awq or self._scale_estimation or self._lora_correction or self._gptq
-        if data_aware_mixed_precision or data_aware_compression:
-            statistics = self._collect_statistics(dataset, nodes_to_compress, graph, model)
+
+        if self._data_aware_mixed_precision or self._data_aware_compression:
+            matmul_nodes_to_compress = self.get_matmul_nodes(nodes_to_compress)
+            matmul_input_to_output_nodes_map = self.get_matmul_input_to_output_nodes_map(
+                matmul_nodes_to_compress, graph
+            )
+            if not is_statistics_provided:
```

Comment:
It makes a lot of sense! I created a ticket 155969.
Let's think about an extension of Algorithm interface
I think that it can be done with caching extension logic for `Torch` or other algorithms